operators/src/processing/meteosat/temperature.rs

```diff
@@ -0,0 +1,637 @@
+        };
+
+        Ok(initialized_operator.boxed())
+    }
+}
+
+impl InitializedRasterOperator for InitializedTemperature {
+    fn result_descriptor(&self) -> &RasterResultDescriptor {
+        &self.result_descriptor
+    }
+
+    fn query_processor(&self) -> Result<TypedRasterQueryProcessor, Error> {
+        let q = self.source.query_processor()?;
+
+        Ok(match q {
+            TypedRasterQueryProcessor::U8(p) => {
+                QueryProcessorOut(Box::new(TemperatureProcessor::new(p, self.params.clone())))
+            }
+            TypedRasterQueryProcessor::U16(p) => {
+                QueryProcessorOut(Box::new(TemperatureProcessor::new(p, self.params.clone())))
+            }
+            TypedRasterQueryProcessor::U32(p) => {
+                QueryProcessorOut(Box::new(TemperatureProcessor::new(p, self.params.clone())))
+            }
+            TypedRasterQueryProcessor::U64(p) => {
+                QueryProcessorOut(Box::new(TemperatureProcessor::new(p, self.params.clone())))
+            }
+            TypedRasterQueryProcessor::I8(p) => {
+                QueryProcessorOut(Box::new(TemperatureProcessor::new(p, self.params.clone())))
+            }
+            TypedRasterQueryProcessor::I16(p) => {
+                QueryProcessorOut(Box::new(TemperatureProcessor::new(p, self.params.clone())))
+            }
+            TypedRasterQueryProcessor::I32(p) => {
+                QueryProcessorOut(Box::new(TemperatureProcessor::new(p, self.params.clone())))
+            }
+            TypedRasterQueryProcessor::I64(p) => {
+                QueryProcessorOut(Box::new(TemperatureProcessor::new(p, self.params.clone())))
+            }
+            TypedRasterQueryProcessor::F32(p) => {
+                QueryProcessorOut(Box::new(TemperatureProcessor::new(p, self.params.clone())))
+            }
+            TypedRasterQueryProcessor::F64(p) => {
+                QueryProcessorOut(Box::new(TemperatureProcessor::new(p, self.params.clone())))
+            }
+        })
+    }
+}
+
+struct TemperatureProcessor<Q, P>
+where
+    Q: RasterQueryProcessor<RasterType = P>,
+{
+    source: Q,
+    params: TemperatureParams,
+    satellite_key: RasterPropertiesKey,
+    channel_key: RasterPropertiesKey,
+    offset_key: RasterPropertiesKey,
+    slope_key: RasterPropertiesKey,
+}
+
+impl<Q, P> TemperatureProcessor<Q, P>
+where
+    Q: RasterQueryProcessor<RasterType = P>,
+{
+    pub fn new(source: Q, params: TemperatureParams) -> Self {
+        Self {
+            source,
+            params,
+            satellite_key: RasterPropertiesKey {
+                domain: Some("msg".into()),
+                key: "Satellite".into(),
+            },
+            channel_key: RasterPropertiesKey {
+                domain: Some("msg".into()),
+                key: "Channel".into(),
+            },
+            offset_key: RasterPropertiesKey {
+                domain: Some("msg".into()),
+                key: "CalibrationOffset".into(),
```

Comment:
we can use offset and slope for this. the source can rename / map the properties.

Reply:
I do not understand :D Simply change the key to  "offset"/"slope"?

Reply:
I added the mapping;